### ISSUE TITLE
add "ignoreComments" option to max-len rule

### ISSUE THREAD

**Issue by zpao** [eslint/eslint#2221]
Several times I've included references to URLs in comments (often when referencing browser quirks). Being URLs, they don't always like to fit in 80 characters, As a result we've set our max-len rule to be a warning. These still show up so that people can fix their code but leaving a few warnings always for long comments is annoying. People come along and try various hacks to get our line length down. I don't want bitly links in my code.

We've also hit this a couple times with JSDoc where our types have long names and we allow multiple types `@param {SomethingReallyLongHere|SomethingElseLong} arg no more space`.

At the end of the day I don't care about the length of any comment lines so I'd like to disable checking those lines.

I'm happy to make this work (I already hacked it together locally). If there's no interest then I'll just distribute as a plugin.

**Comment by nzakas** [comment]
Seems like a nice option to have. It should be false by default to maintain current behavior.

**Comment by lo1tuma** [comment]
Should this rule only be applied for comments which appear on a separate line?

What about the following examples:
```js
var aReallyLongVariableNameThatDescribesSomething = "a really long string literal"; // any comment
```

```js
var foo = /* a really long comment describing a workaround and contains a link to a log url http://github.com*/ "bar";
```

Related issue #1661.

**Comment by nzakas** [comment]
Good question. I think I'd start with comments on their own line.

**Comment by vvo** [comment]
Nice idea
